src/readers/src/TRM.h

```diff
@@ -41,15 +50,16 @@ struct Coordinates {
 
 class BaseLayer {
  public:
+	BaseLayer(LayerType type) { type_ = type; }
 	// getters
 	[[nodiscard]] LayerType type() const;
-  [[nodiscard]] std::string_view type_tag() const;
-  [[nodiscard]] float thermal_conductivity() const;
-  [[nodiscard]] float thickness() const;
+	[[nodiscard]] std::string_view type_tag() const;
+	[[nodiscard]] float thermal_conductivity() const;
+	[[nodiscard]] float thickness() const;
 
 	// read ryabov file content from stream
 	virtual std::istream& read(std::istream& in) = 0;
-	virtual GeomStorage<BasicShape> geometry() = 0;
+	virtual GeomStorage<BasicShape> geometry()   = 0;
```

Comment:
We are using only bounding box from shape in virtex data provider and always iterate over all shapes, why can't we use vector of bounding boxes or other simplier structures for storing?

Reply:
For future PR